core/src/main/kotlin/utilities/json.kt

```diff
@@ -0,0 +1,69 @@
+package org.jetbrains.dokka.utilities
+
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.SerializerProvider
+import com.fasterxml.jackson.databind.deser.std.StdScalarDeserializer
+import com.fasterxml.jackson.databind.module.SimpleModule
+import com.fasterxml.jackson.databind.ser.std.StdScalarSerializer
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.jacksonTypeRef
+import org.jetbrains.dokka.DokkaConfiguration.SourceRoot
+import org.jetbrains.dokka.SourceRootImpl
+import java.io.File
+import com.fasterxml.jackson.core.type.TypeReference as JacksonTypeReference
+
+private val objectMapper = run {
+    val module = SimpleModule().apply {
+        addSerializer(FileSerializer)
+        addSerializer(SourceRoot::class.java, SourceRootSerializer)
+        addDeserializer(SourceRootImpl::class.java, SourceRootImplDeserializer)
+        addDeserializer(SourceRoot::class.java, SourceRootImplDeserializer)
+    }
+    jacksonObjectMapper()
+        .registerModule(module)
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+}
```

Comment:
IMHO it would be more readable when wrote like this:
```kotlin
private val objectMapper = SimpleModule().apply {
... }.let { module ->
    jacksonObjectMapper()
        .registerModule(module)
        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
}
```
but maybe that's only my preference

Reply:
Hihi, I actually think the opposite: There is no reason for scoping, so I only see using a scoping function as making it even harder to parse (since scopes are expensive for me to parse in my head 🙄 )